docs/porting/libraries.md

```diff
@@ -1,5 +1,151 @@
+
+2. If you have assemblies that are mostly incompatible, is it worth leaving them on .NET Framework for now?  Note that this will depend heavily on the architecture of your system.
+
+   a. You may have some functionality in your library that's incompatible with .NET Core because it relies too heavily on .NET Framework or Windows-specific functionality.  Is it worth leaving that functionality behind for now and releasing a .NET Core verson of your library with less features for the time being?
+
+3. Are there alternative .NET Core APIs for incompatible .NET Framework APIs you're using?
+
+   a. For example, if you're using `WebClient` you may be able to use `HttpClient` instead.
+   
+   b. You may also consider using a slightly different .NET Core API to do some work even if it's not a drop-in replacement.  For example, if you don't require XML Schema discovery, you could use Linq to Xml to parse XML data instead of using XMLSchema to handle the parsing.
+   
+4. Is it reasonable to write some code a .NET Framework API was doing yourself?
+
+   a. Consider modifying and using code from the [.NET Framework Reference Source](referencesource.microsoft.com).  It's licensed under the [MIT License](https://github.com/Microsoft/referencesource/blob/master/LICENSE.txt) so you have significant freedom in doing this.
+   
+This analysis phase could take some time.  You may find that you'll have to refactor your code or make significant changes to how you perform certain tasks.  You may even come to the conlcusion that porting to .NET Core isn't feasible at this time.
+
+## Choosing the project system and getting a basis for compilation
+
+Once you have analyzed the ApiPort report and your codebase, the next step is to actually begin porting your code!
+
+### Making a choice: xproj or Portable Class Library?
+
+The first thing you'll need to do is decide on the project system you want to use.  You have two choices: `xproj` or `csproj` via a Portable Class Library (PCL).  Deciding which to pick comes down to one simple question:
+
+Do you have any need to support anything more than just having Visual Studio recognize your project?  Examples of this may be localized resources in a `.resx` folder, working seamlessly with a custom build process, and others.
+
+If yes, you should create a new PCL and select .NET Core as the target.  If you have no needs other than building and running your code in Visual Studio, you can select an xproj project type.
+
+### Getting a basis for compilation
+
+Once you've picked your project type and created a new Class Library, you'll notice that there's a new file: `project.json`.  This is the file that handles many things about your project - stating dependencies, configurations, supported target frameworks, etc. - and is used to generate a `project.lock.json` file that is used by the compiler and build system to build your project.  You can read more about `project.json` TODO: LINK_HERE.
```

Comment:
Yeah, I agree.  I've changed this up a bit to introduce it as the result of running a restore.